lib/fs/memfs/memfs.c

```diff
@@ -253,6 +253,42 @@ static ssize_t memfs_read(filecookie *fcookie, void *buf, off_t off, size_t len)
     return len;
 }
 
+static status_t memfs_truncate(filecookie *fcookie, uint64_t len)
+{
+    LTRACEF("filecookie %p, len %llu\n", fcookie, len);
+
+    status_t rc = NO_ERROR;
+
+    memfs_file_t *file = (memfs_file_t *)fcookie;
+
+    mutex_acquire(&file->fs->lock);
+
+    // Can't use truncate to grow a file.
+    if (len > file->len)
+        return ERR_INVALID_ARGS;
+
+    void *ptr = realloc(file->ptr, len);
```

Comment:
Pretty sure all the reallocs are fine and ,0 will produce a null... but then the line before will fail.